### ISSUE TITLE
Value mismatch in body with nested json

### ISSUE THREAD

**Issue by varayal** [taverntesting/tavern#265]
Based on below response I'm trying to validate the **first_name: "Eve"** but I'm confused about how to do it. I checked the documentation twice and did not found samples to help me out.

```json
{
    "page": 2,
    "per_page": 3,
    "total": 12,
    "total_pages": 4,
    "data": [
        {
            "id": 4,
            "email": "eve.holt@reqres.in",
            "first_name": "Eve",
            "last_name": "Holt",
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
        },
        {
            "id": 5,
            "email": "charles.morris@reqres.in",
            "first_name": "Charles",
            "last_name": "Morris",
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
        },
        {
            "id": 6,
            "email": "tracey.ramos@reqres.in",
            "first_name": "Tracey",
            "last_name": "Ramos",
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
        }
    ]
}
```
See below my yaml test

```
Test_name: test01

stages:
  - name: getting the list of users
    request:
      url: http://localhost:5000/api/users?page=2
      method: GET  
    response:
      status_code: 200
      body:
        page: 2
        per_page: 3
        data:
          last_name: 'Eve'
```

I got the following error

`base.py                     37 ERROR    Value mismatch in body: Type of returned data was different than expected (expected["data"] = '{'last_name': 'Holt'}' (type = <class 'dict'>), actual["data"] = '[{'id': 4, 'email': 'eve.holt@reqres.in', 'first_name': 'Eve', 'last_name': 'Holt', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg'}, {'id': 5, 'email': 'charles.morris@reqres.in', 'first_name': 'Charles', 'last_name': 'Morris', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg'}, {'id': 6, 'email': 'tracey.ramos@reqres.in', 'first_name': 'Tracey', 'last_name': 'Ramos', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg'}]' (type = <class 'list'>))`

Any thoughts?

**Comment by varayal** [comment]
Still facing the issue about validating the email based on the below-nested JSON

```
 {
"data": {
"id": 4,
"email": "eve.holt@reqres.in",
"first_name": "Eve",
"last_name": "Holt",
"avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
}
}
```

**Comment by michaelboulton** [comment]
When you say it doesn't work, you mean it gives you an error about the response not matching?

You will either need to specify the entire json response or use the `strict: False` flag in the stage/test (see the [documentation](https://tavern.readthedocs.io/en/latest/basics.html#strict-key-checking) ). Something like:


 ```yaml
  - name: Check for email
    strict: False  # <--
    request:
      ...
    response:
      status_code: 200
      body:
        page: 2
        per_page: 3
        data:
          email:  "eve.holt@reqres.in"
```

**Comment by MonRes** [comment]
Hi, I use strict: False but I still got this ERROR: Value mismatch in body: Length of returned list was different than expected.

This is my yaml:

    response:
      strict: False
      status_code: 200
      body:
        - measurements:
          - name: temperature
            value: 12.4

**Comment by michaelboulton** [comment]
@MonRes Could you provide an example of the response you would expect from that? It looks like it might be a bug in Tavern

**Comment by varayal** [comment]
This could be related to this issue as well https://github.com/taverntesting/tavern/issues/303

**Comment by MonRes** [comment]
@michaelboulton sure. This is my response:

[
"measurements": [{
"name": temperature,
"value": 12.4,
"timestamp": 2019-03-13}, 
{"name": temperature,
"value": 12.7,
"timestamp": 2019-03-13}],
"measurements": [{
"name": temperature,
"value": 12,
"timestamp": 2019-03-12}, 
{"name": temperature,
"value": 12.1,
"timestamp": 2019-03-12}]
]

@varayal could be...

**Comment by michaelboulton** [comment]
@MonRes I think I misread your original question, what you're trying to do is https://github.com/taverntesting/tavern/issues/133 which you can't currently do, see https://github.com/taverntesting/tavern/pull/162

**Comment by michaelboulton** [comment]
Just going to close the issues related to selecting items in a nested list because it is a known problem which we will fix in 1.0 - going to keep discussion in this issue https://github.com/taverntesting/tavern/issues/133#issuecomment-481774309
